x/ecocredit/server/core/create_batch_test.go

```diff
@@ -0,0 +1,133 @@
+package core
+
+import (
+	"context"
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/golang/mock/gomock"
+	ecocreditv1beta1 "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1beta1"
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+	"github.com/regen-network/regen-ledger/x/ecocredit/v1beta1"
+	"gotest.tools/v3/assert"
+	"testing"
+	"time"
+)
+
+func TestCreateBatch_Valid(t *testing.T) {
+	t.Parallel()
+	s := setupBase(t)
+	batchTestSetup(t, s.ctx, s.stateStore, s.addr)
+
+	any := gomock.Any()
+	s.paramsKeeper.EXPECT().GetParamSet(any, any).Do(func(any interface{}, p *ecocredit.Params) {
+		p.AllowlistEnabled = false
+		p.CreditClassFee = types.NewCoins(types.NewInt64Coin("foo", 20))
+		p.CreditTypes = []*ecocredit.CreditType{{Name: "carbon", Abbreviation: "C", Unit: "tonne", Precision: 6}}
+	}).AnyTimes()
```

Comment:
Let's be specific if we include a times check.
```suggestion
	}).Times(1)
```